serdes-native/src/main/java/io/ballerina/stdlib/serdes/Deserializer.java

```diff
@@ -0,0 +1,360 @@
+
+                } else {
+                    Map<String, Object> nestedMap = recordToBallerina(nestedDynamicMessage, type, schema);
+                    String recordTypeName = getRecordTypeName(type, fieldName);
+
+                    BMap<BString, Object> nestedRecord = ValueCreator.createRecordValue(
+                            type.getPackage(),
+                            recordTypeName,
+                            nestedMap
+                    );
+
+                    map.put(fieldName, nestedRecord);
+                }
+            } else if (value.getClass().getSimpleName().equals(BYTE) || entry.getKey().isRepeated()) {
+                if (!value.getClass().getSimpleName().equals(BYTE)) {
+                    Type elementType = getArrayElementType(type, fieldName);
+                    Object handleArray = arrayToBallerina(value, elementType, schema, 1);
+
+                    map.put(fieldName, handleArray);
+                } else {
+                    Object handleArray = arrayToBallerina(value, type, schema, 1);
+
+                    map.put(fieldName, handleArray);
+                }
+
+            } else if (DataTypeMapper.getProtoTypeFromJavaType(value.getClass().getSimpleName()) != null) {
+                Object handlePrimitive = primitiveToBallerina(value);
+
+                map.put(fieldName, handlePrimitive);
+            } else {
+                throw createSerdesError(UNSUPPORTED_DATA_TYPE + value.getClass().getSimpleName(), SERDES_ERROR);
+            }
+        }
+
+        return map;
+    }
+
+    private static String getRecordTypeName(Type type, String fieldName) {
+        RecordType recordType = (RecordType) type;
+
+        for (Map.Entry<String, Field> entry: recordType.getFields().entrySet()) {
+            Type fieldType = entry.getValue().getFieldType();
+
+            if (fieldType.getTag() == TypeTags.RECORD_TYPE_TAG && entry.getKey().equals(fieldName)) {
+                return fieldType.getName();
+            } else if (fieldType.getTag() == TypeTags.RECORD_TYPE_TAG) {
+                return getRecordTypeName(fieldType, fieldName);
+            } else {
+                continue;
+            }
+        }
+
+        return null;
+    }
+
+    private static Type getArrayElementType(Type type, String fieldName) {
+        RecordType recordType = (RecordType) type;
+
+        for (Map.Entry<String, Field> entry: recordType.getFields().entrySet()) {
+            if (entry.getValue().getFieldType().getTag() == TypeTags.ARRAY_TAG && entry.getKey().equals(fieldName)) {
+                ArrayType arrayType = (ArrayType) entry.getValue().getFieldType();
+
+                return arrayType.getElementType();
+            } else if (entry.getValue().getFieldType().getTag() == TypeTags.RECORD_TYPE_TAG) {
+                return getArrayElementType(entry.getValue().getFieldType(), fieldName);
+            }
+        }
+
+        return null;
```

Comment:
Is returning `null` is the correct way to do this?

Reply:
Fixed